GillianCore/GIL_Syntax/Literal.ml

```diff
@@ -26,7 +31,7 @@ let rec pp fmt x =
   | Constant c -> Fmt.string fmt (Constant.str c)
   | Bool b     -> if b then Fmt.string fmt "true" else Fmt.string fmt "false"
   | Int i      -> Fmt.int fmt i
-  | Num n      -> Fmt.float fmt n
+  | Num n      -> Fmt.pf fmt "%s" (remove_trailing_dot (Printf.sprintf "%F" n))
```

Comment:
Wait I don't get that, why did you remove the trailing dot ? We're about to add proper integers to gillian, and when we do, we need the trailing dot to differentiate between floats and integers
Also, right now, the regexp for floats is
```ocaml
let float = '-'? digit+ ('.' digit*)?
```
which is compatible with a trailing dot
Sorry, I think you explained earlier but I didn't get it

Reply:
Okay, I thought it was a bit redundant when there are no digits after the decimal points (or only zeros). I'll change it to use `%F`.